vcd/resource_vcd_network_routed.go

```diff
@@ -0,0 +1,245 @@
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+
+			"dhcp_pool": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"start_address": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"end_address": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"default_lease_time": &schema.Schema{
+							Type:     schema.TypeInt,
+							Default:  3600,
+							Optional: true,
+						},
+
+						"max_lease_time": &schema.Schema{
+							Type:     schema.TypeInt,
+							Default:  7200,
+							Optional: true,
+						},
+					},
+				},
+				Set: resourceVcdNetworkIPAddressHash,
+			},
+			"static_ip_pool": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"start_address": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"end_address": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceVcdNetworkIPAddressHash,
+			},
+		},
+	}
+}
+
+func resourceVcdNetworkRoutedCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] CLIENT: %#v", vcdClient)
+	vcdClient.Mutex.Lock()
+	defer vcdClient.Mutex.Unlock()
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	// Collecting components
+	egName, ok := d.GetOk("edge_gateway")
+	var edgeGateway govcd.EdgeGateway
+
+	edgeGatewayName := ""
+	if ok {
+		edgeGatewayName = egName.(string)
+		eg, err := vdc.FindEdgeGateway(edgeGatewayName)
+		if err == nil {
+			edgeGateway = eg
+		} else {
+			return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+		}
+	}
+
+	gwName, ok := d.GetOk("gateway")
+	gatewayName := ""
+	if ok {
+		gatewayName = gwName.(string)
+	}
+
+	ipRanges := expandIPRange(d.Get("static_ip_pool").(*schema.Set).List())
+
+	orgVDCNetwork := &types.OrgVDCNetwork{
+		Xmlns: "http://www.vmware.com/vcloud/v1.5",
+		Name:  d.Get("name").(string),
+
+		EdgeGateway: &types.Reference{
+			HREF: edgeGateway.EdgeGateway.HREF,
+		},
+		Configuration: &types.NetworkConfiguration{
+			FenceMode: "natRouted",
+			IPScopes: &types.IPScopes{
+				IPScope: types.IPScope{
+					IsInherited: false,
+					Gateway:     gatewayName,
+					Netmask:     d.Get("netmask").(string),
+					DNS1:        d.Get("dns1").(string),
+					DNS2:        d.Get("dns2").(string),
+					DNSSuffix:   d.Get("dns_suffix").(string),
+					IPRanges:    &ipRanges,
+				},
+			},
+			BackwardCompatibilityMode: true,
+		},
+		IsShared: d.Get("shared").(bool),
+	}
+
+	util.Logger.Printf("[INFO] NETWORK: %#v", orgVDCNetwork)
```

Comment:
This will look like random messages in info logs with strange information about what happend. "NETWORK:". Another question do we really want this in info level.